src/jshint.js

```diff
@@ -3562,6 +3576,10 @@ var JSHINT = (function() {
         tokens = destructuringExpression();
         lone = false;
       } else {
+        if (inexport) {
+          exported[state.tokens.next.value] = true;
+          state.tokens.next.exported = true;
+        }
```

Comment:
setting the exported flag should happen after the call to `identifier()` below (which will make sure that it's actually an identifier)